gff3tool/bin/gff3_sort.py

```diff
@@ -11,30 +11,50 @@
 from gff3tool.lib.gff3 import Gff3
 from gff3tool.bin import version
 
-__version__ = version.__version__
 
+__version__ = version.__version__
 
-def PositionSort(linelist):
+def PositionSort(linelist,reference):
     # the input argument, 'linelist', is a python list collecting all the features you would like to sort by genomic coordinates
     id2line = {}
     id2start = {}
     seq2id = {}
-    for line in linelist:
-        id2line[str(line['line_raw'])] = line
-        id2start[str(line['line_raw'])] = (line['start'],line['line_index'])
-        tmp = re.search('(.+?)(\d+)',line['seqid']) # Truncate the sequence ID, and only keep the sequence ID number
-        try:
-            seqnum = tmp.groups()[1]
-        except AttributeError:
-            print('ERROR  [Missing SeqID] Missing SeqID.\n\t\t- Line {0:s}: {1:s}'.format(str(line['line_index']+1),line['line_raw']))
-            sys.exit(1)
-        # 'seq2id': a dictionary mapping sequence number to their features
-        if seq2id.has_key(seqnum):
-            seq2id[seqnum].append(str(line['line_raw']))
-        else:
-            seq2id[seqnum] = [str(line['line_raw'])]
-    # Sort by sequence ID number, and store them in 'keys'
-    keys = sorted(seq2id, key=lambda i: int(i))
+    if reference==True:
+        #print('New_Function')
+        for line in linelist:
+            id2line[str(line['line_raw'])] = line
+            id2start[str(line['line_raw'])] = (line['start'],line['line_index'])
+            tmp = re.search('(.+?)(\D+)',line['seqid']) # Truncate the sequence ID, and only keep the sequence ID number
```

Comment:
@tony006469 can you modify this so the regex can capture any character but a tab? (Ideally you'd split the line on tabs and get the first element of the array). For more info about the seqid requirements, you can look at https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md